### DISCUSSION TITLE
Easily rename accepted internal link suggestion.

### DISCUSSION THREAD

**Discussion by lukasschulte03** [tadashi-aikawa/obsidian-various-complements-plugin#273] [Ideas]
First off, this plugin is amazing, and I use it everyday, especially its internal link complement features.

But I have the issue that I, most of the time I want to rename my internal links, since they're almost always either incorrectly capitalized, or inflected incorrectly, or so on...
This becomes quite a hassle after a while. Always having to go back a couple of characters, insert a pipe character, write the actual display name, and go forward again.

I was wondering if there's any way to make a feature, where every time you accept an autocomplete suggestion to an internal link, an input prompt pops up where you can insert if you want the display name to be different that of the underlying notes name (or alias that you've referred to). If you don't want a different display name, just hit enter again and leave it empty, but if you type something and hit enter, the display name should be changed to that.

Of course it should be toggleable in the settings, and off by default, since I don't see this fitting the needs of the majority of users.

I would find this much easier than alias managing since that's not really applicable in my case. This solution feels much easier when the alternative is to make 10's of aliases for each note, which would suck...

*(Maybe there's already a good solution for this and I'm just ignorant, please inform me that case.)*

**Comment by tadashi-aikawa** [comment]
Hi, @lukasschulte03 :)

Thank you for your very interesting opinion. I strongly agree and am motivated to implement a prototype.

> an input prompt pops up where you can insert if you want the display name to be different that of the underlying notes name

I also think that initially taking an approach to display an input popup is a good idea. I'll aim to start with this.

**Comment by tadashi-aikawa** [comment]
I will have enough time this weekend to make progress on the development 👍

**Comment by tadashi-aikawa** [comment]
@lukasschulte03 
https://github.com/tadashi-aikawa/obsidian-various-complements-plugin/issues/282#issuecomment-1936915195
